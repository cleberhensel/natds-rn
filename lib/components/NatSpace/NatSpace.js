import React from 'react';
import PropTypes from 'prop-types';

import { NatView } from './NatSpace.styles';

export const NatSpace = ({
  children,
  marginTop,
  marginBottom,
  marginRight,
  marginLeft,
  paddingTop,
  paddingBottom,
  paddingRight,
  paddingLeft,
  backgroundColor,
}) => (
  <NatView {...{
    marginTop,
    marginBottom,
    marginRight,
    marginLeft,
    paddingTop,
    paddingBottom,
    paddingRight,
    paddingLeft,
    backgroundColor,
  }}
  >
    { children }
  </NatView>
);

NatSpace.propTypes = {
  children: PropTypes.node.isRequired,
  marginTop: PropTypes.string,
  marginBottom: PropTypes.string,
  marginRight: PropTypes.string,
  marginLeft: PropTypes.string,
  paddingTop: PropTypes.string,
  paddingBottom: PropTypes.string,
  paddingRight: PropTypes.string,
  paddingLeft: PropTypes.string,
  backgroundColor: PropTypes.string,
};

NatSpace.defaultProps = {
  marginTop: 'none',
  marginBottom: 'small',
  marginRight: 'none',
  marginLeft: 'none',
  paddingTop: 'none',
  paddingBottom: 'none',
  paddingRight: 'none',
  paddingLeft: 'none',
  backgroundColor: 'transparent',
};
