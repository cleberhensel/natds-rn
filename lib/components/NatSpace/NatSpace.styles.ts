import styled from 'styled-components/native';
import { sizes } from '../../tokens/sizes';

export const NatView = styled.View`
  margin-top: ${props => sizes[props.marginTop]}px;
  margin-bottom: ${props => sizes[props.marginBottom]}px;
  margin-right: ${props => sizes[props.marginRight]}px;
  margin-left: ${props => sizes[props.marginLeft]}px;
  padding-top: ${props => sizes[props.paddingTop]}px;
  padding-bottom: ${props => sizes[props.paddingBottom]}px;
  padding-right: ${props => sizes[props.paddingRight]}px;
  padding-left: ${props => sizes[props.paddingLeft]}px;
  background-color: ${props => (props.backgroundColor ? props.backgroundColor : 'none')};
`;
