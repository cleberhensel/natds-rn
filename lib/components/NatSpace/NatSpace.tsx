import React from "react";
import { NatView } from "./NatSpace.styles";
type NatSpaceProps = {
  backgroundColor?: string,
  marginBottom?: string,
  marginLeft?: string,
  marginRight?: string,
  marginTop?: string,
  paddingBottom?: string,
  paddingLeft?: string,
  paddingRight?: string,
  paddingTop?: string
};
export const NatSpace: React.SFC<NatSpaceProps> = ({
  children,
  marginTop,
  marginBottom,
  marginRight,
  marginLeft,
  paddingTop,
  paddingBottom,
  paddingRight,
  paddingLeft,
  backgroundColor
}) => (
  <NatView
    {...{
      backgroundColor,
      marginBottom,
      marginLeft,
      marginRight,
      marginTop,
      paddingBottom,
      paddingLeft,
      paddingRight,
      paddingTop
    }}
  >
    {children}
  </NatView>
);
NatSpace.defaultProps = {
  backgroundColor: "transparent",
  marginBottom: "small",
  marginLeft: "none",
  marginRight: "none",
  marginTop: "none",
  paddingBottom: "none",
  paddingLeft: "none",
  paddingRight: "none",
  paddingTop: "none"
};
