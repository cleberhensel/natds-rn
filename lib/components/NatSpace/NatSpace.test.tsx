import React from "react";
import renderer from "react-test-renderer";
import { Text } from "react-native";
import "jest-styled-components/native";
import { NatSpace } from "./NatSpace";
describe("NatSpace component", () => {
  it("Should render Space component correctly", () => {
    const space = renderer
      .create(
        <NatSpace>
          <Text>Hello</Text>
        </NatSpace>
      )
      .toJSON();
    expect(space).toMatchSnapshot("NatSpace snapshot");
  });
});
