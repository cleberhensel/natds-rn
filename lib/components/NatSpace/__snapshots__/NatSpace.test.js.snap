// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NatSpace component Should render Space component correctly: NatSpace snapshot 1`] = `
<View
  backgroundColor="transparent"
  marginBottom="small"
  marginLeft="none"
  marginRight="none"
  marginTop="none"
  paddingBottom="none"
  paddingLeft="none"
  paddingRight="none"
  paddingTop="none"
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "marginBottom": 16,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      },
    ]
  }
>
  <Text>
    Hello
  </Text>
</View>
`;
